### PR TITLE
feat: `usort` hints integration tests 

### PR DESCRIPTION
**Pull request by TAdev0**
@cicr99  i checked hints separately locally and it seems fine, except  `usortBodyCode` which is not detected by our VM, just like `signedDivRem`

We might need to do like lambda go and declare each hint as a const

### CODE REVIEW

**Comment by TAdev0** [comment]
@cicr99  @har777  @MaksymMalicki 

Usort integration test is finished. I modified a few things in usort hints:
- There was fp.Element used in 2 places where uint64 should have been used. It saves some computation moreover
- I hardcoded usort_max_size which is a global(), just like we did for keccak state size or keccak max input size
- I reorder all the usort hints so that they are in the order they are executed in `usort` cairo function

Let's wait Finalize Segment to be merged to make sure the memory traces match, but i tested outputs and everything against other PR and its ok

**Comment by TAdev0** [comment]
@har777  @cicr99  @MaksymMalicki   ready for approval/merging

I added a new test for it to make sure multiplicities works well (lambdaclass didnt cover this)

**Review by Sh0g0-1758 — COMMENTED** [review]
Well the integration test looks good. Other than that I believe its mostly relocation with a change in one place where you have converted felt to uint64. I think @cicr99 should have a take on that. I just left one comment on the change in implementation of a hint.

File: pkg/hintrunner/zero/zerohint_usort.go [hunk 1]
```diff
@@ -12,33 +12,30 @@ import (
...
-	if err != nil {
-		return nil, err
-	}
+// UsortEnterScope hint enters a new scope with `__usort_max_size` value
+//
+// `newUsortEnterScopeHint` doesn't take any operander as argument
+//
+// `newUsortEnterScopeHint` gets `__usort_max_size` value from the current
+// scope and enters a new scope with this same value
+func newUsortEnterScopeHint() hinter.Hinter {
```
[the remark this answers is not in the document]
Comment (TAdev0) [reply pkg/hintrunner/zero/zerohint_usort.go:21]: @Sh0g0-1758  we decided to hardcode `__usort_max_size` and make it always available. i'm waiting for news from the cairo vm rust team to know how they handle this. But for now we leave it like this

**Comment by TAdev0** [comment]
@Sh0g0-1758  we already discussed with Carmen the change to uint64 here. If you check in detail, you'll see i changed it always for variable that represents either a length of array or a index in an array. Its ok then

File: pkg/hintrunner/zero/zerohint_usort.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Sh0g0-1758) [reply pkg/hintrunner/zero/zerohint_usort.go:21]: okay!

**Review by Sh0g0-1758 — APPROVED** [review]
LGTM

**Pull request merged**